sklearn/linear_model/tests/test_bayes.py

```diff
@@ -56,3 +56,55 @@ def test_toy_ard_object():
     # Check that the model could approximately learn the identity function
     test = [[1], [3], [4]]
     assert_array_almost_equal(clf.predict(test), [1, 3, 4], 2)
+
+
+def test_return_std_bayesian():
+    def f(X):
+        return np.dot(X, w) + b
+
+    def f_noise(X, noise_mult):
+        return f(X) + np.random.randn(X.shape[0])*noise_mult
+
+    d = 5
+    n_train = 50
+    n_test = 10
+
+    noise_mult = 0.1
+    w = np.array([1.0, 0.0, 1.0, -1.0, 0.0])
+    b = 1.0
+
+    X = np.random.random((n_train, d))
+    X_test = np.random.random((n_test, d))
+    y = f_noise(X, noise_mult)
+
+    m1 = BayesianRidge()
+    m1.fit(X, y)
+    X_test = np.random.random((n_test, d))
+    y_mean, y_std = m1.predict(X_test, return_std=True)
+    assert_array_almost_equal(y_std, noise_mult, decimal=1)
+
+
+def test_return_std_ard():
+    def f(X):
+        return np.dot(X, w) + b
+
+    def f_noise(X, noise_mult):
+        return f(X) + np.random.randn(X.shape[0])*noise_mult
+
+    d = 5
+    n_train = 50
+    n_test = 10
+
+    noise_mult = 0.1
```

Comment:
Works fine for Bayesian Ridge, but unfortunately, ARD behaves oddly because it gets rid of a bunch of dimensions. It gets it MOSTLY right. If you set `noise_mult = 1.0`, ARD will get `1.2` or `1.1`, which is good enough for the estimation of standard deviation of noise, but I'd have to bump up `decimal=0` for it to still pass the tests for such a large noise. I'll do that for the next commit unless you have another suggestion.